Cognite.Jetfire.Cli/Transform/TransformCommand.cs

```diff
@@ -0,0 +1,205 @@
+                    Description = "Maximum amount of time to wait for job to complete"
+                },
+                new Option<TimeSpan>("--interval", () => TimeSpan.FromSeconds(5))
+                {
+                    Description = "Time interval for polling job status"
+                }
+            };
+            Command.Handler = CommandHandler.Create(
+                (Func<IConsole, string, int?, string, bool, bool, TimeSpan, TimeSpan, Task<int>>)Handle
+            );
+        }
+
+        ISecretsProvider secrets;
+
+        ManifestParser parser = new ManifestParser();
+
+        public Command Command { get; }
+
+        async Task<int> Handle(
+            IConsole console,
+            string cluster,
+            int? id,
+            string externalId,
+            bool watch,
+            bool watchOnly,
+            TimeSpan timeout,
+            TimeSpan interval
+        )
+        {
+            var client = JetfireClientFactory.CreateClient(secrets, cluster);
+
+            if (id == null && externalId == null ||
+                id != null && externalId != null)
+            {
+                throw new Exception("Either --id or --external-id must be specified");
+            }
+
+            int configId;
+            if (id != null)
+            {
+                configId = id.Value;
+            }
+            else
+            {
+                var config = await client.TransformConfigByExternalId(externalId);
+                configId = config.Id;
+            }
+
+            string jobUuid;
+            if (watchOnly)
+            {
+                var jobs = await client.TransformConfigRecentJobs(configId);
+                jobUuid = jobs.First().Uuid;
+            }
+            else
+            {
+                console.Out.Write($"Starting job for transformation {configId}");
+                if (externalId != null)
+                {
+                    console.Out.Write($" (externalId: {externalId})");
+                }
+                console.Out.WriteLine();
+
+                var startJobResult = await client.TransformConfigStartJob(configId);
+                console.Out.WriteLine($"Started job {startJobResult.JobId}");
+                jobUuid = startJobResult.JobId;
+            }
+
+            if (watch || watchOnly)
+            {
+                console.Out.WriteLine($"Watching job {jobUuid} for completion, timing out after {timeout}");
+
+                var cancellationTokenSource = new CancellationTokenSource();
+                var cancellationToken = cancellationTokenSource.Token;
+                try
+                {
+                    cancellationTokenSource.CancelAfter(timeout);
+
+                    while (!await IsJobCompleted(client, configId, jobUuid, cancellationToken))
+                    {
+                        await Task.Delay(interval, cancellationToken);
+                    }
+                    Console.Out.WriteLine("Job completed successfully");
```

Comment:
This should be using `console` like the others, I think? Or is it intentionally using `Console` instead for the successful case?

Reply:
Ah, good catch! That's def. an oversight.